### DISCUSSION TITLE
Add suport for color and icon on branches and nodes

### DISCUSSION THREAD

**Discussion by RrR2010** [markmap/markmap#105] [Ideas]
Hi!

It would be awesome if you can add suport for adding colors and icons (maybe like CSS tags) to branches and nodes.
It can be something like this:

#Markmap
- Node with a Esmerald color text and Silver background color {background-color: #BDC3C7, color: #2ECC71}
- Node with default theme colors, Turquoise branch color and smile emoji icon {border-color: #1ABC9C, icon: smile}

**Comment by RrR2010** [comment]
I found that I can utilize html tags to personalize the node's texts.
It can be done things like:
`<span style="font-weight:bold;">Text in bold<br>and with a line break</span> \\ The ** symbols are not actually working for bold`
and the result will be like:

**Text in bold**
and with a line break

Text can be colored to with `<span style="color:red">Colored text</span>` or background with `<span style="background-color:green">Colored text</span>`

It would be nice yet to become possible to edit branch colors too.

**Comment by goeck** [comment]
Coloring the branches on purpose is for definetely a feature that is really important, in order to craft a nice mindmap using this nice technology.

**Comment by gera2ld** [comment]
For icons I think emoji is enough.
Branch colors can be roughly supported via frontmatter, try it at <https://markmap.js.org>.
However the color of a single branch is not supported directly in Markdown, but can be changed via API. (could be supported via magic comment)

I believe Markdown shouldn't be full of magic comments or styles, it puts content first, so if you really want so much customization, maybe you should use API directly.
